numpy/core/src/multiarray/refcount.c

```diff
@@ -258,6 +307,10 @@ _fillobject(char *optr, PyObject *obj, PyArray_Descr *dtype)
             Py_XDECREF(arr);
```

Comment:
Out of scope, but - line 295 will set an exception if an integer larger than `2**64` is passed in.

Reply:
Frankly, I hope that this is only ever used for `0` or `None`, unfortunately it is public API.